shell/browser/ui/file_dialog_linux_portal.cc

```diff
@@ -58,11 +58,7 @@ void CheckPortalAvailabilityOnBusThread() {
   if (flag->IsSet())
     return;
 
-  dbus::Bus::Options options;
-  options.bus_type = dbus::Bus::SESSION;
-  options.connection_type = dbus::Bus::PRIVATE;
-  options.dbus_task_runner = dbus_thread_linux::GetTaskRunner();
-  scoped_refptr<dbus::Bus> bus = base::MakeRefCounted<dbus::Bus>(options);
+  scoped_refptr<dbus::Bus> bus = dbus_thread_linux::GetSharedSessionBus();
```

Comment:
The portal code path cannot use the shared session bus, the upstream implementation refactored the portal code to use shared session bus but it doesn't work for us context in https://github.com/electron/electron/issues/46652#issuecomment-2846088555, hence we explicitly create and mange a bus here.